File: other_languages/python/binary_search.py
import sys
from shared_sort import sorted_array


def main(args):
    try:
        to_find = int(args[0])
        binary_search(sorted_array, to_find)
    except IndexError:
        print("usage: python binary_search.py n")
        exit(1)


def binary_search(sequence, target):
    low = 0
    high = len(sequence) - 1
    while (high >= low):
        mid = (high + low) // 2
        print(f"Checking position {mid}")
        if target < sequence[mid]:
            high = mid - 1
        elif target > sequence[mid]:
            low = mid + 1
        else:
            print(f"{target} found at position {mid}")
            return
    print(f"{target} not found at any position")


if __name__ == '__main__':
    main(sys.argv[1:])

File: other_languages/javascript/binarySearch.js
const sorted_array = [1, 2, 3, 5, 6, 7, 8, 9, 12, 14, 17, 19, 34, 56, 77, 89, 100];
const args = process.argv.slice(2);

function binarySearch (sequence, target) {
    let low = 0;
    let high = sequence.length - 1;
    while (high >= low) {
        let mid = Math.floor((high + low) / 2)
        console.log(`Checking position ${mid}`);
        if (target < sequence[mid]) {
            high = mid - 1;
        } else if (target > sequence[mid]) {
            low = mid + 1;
        } else {
            console.log(`${target} found at position ${mid}`);
            return;
        };
    };
    console.log(`${target} not found at any position`);
}

binarySearch(sorted_array, parseInt(args[0]));
